src/Catalog.Api/Controllers/ItemController.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System.Net;
+using Catalog.Api.Mappers;
+using Catalog.Api.Requests;
+using Catalog.Api.Responses;
+using Catalog.Application.Services;
+using Catalog.Domain.Exceptions;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Swashbuckle.AspNetCore.Annotations;
+
+namespace Catalog.Api.Controllers;
+
+[ApiController]
+[Route("catalog")]
+public class ItemController : ControllerBase
+{
+    private readonly IItemService itemService;
+
+    public ItemController(IItemService itemService)
+    {
+        this.itemService = itemService;
+    }
+
+    [HttpGet]
+    [Route("item/{categoryId}")]
+    [SwaggerResponse((int)HttpStatusCode.OK, nameof(HttpStatusCode.OK), typeof(IEnumerable<ItemResponse>))]
+    [SwaggerResponse((int)HttpStatusCode.InternalServerError, nameof(HttpStatusCode.InternalServerError))]
+    public async Task<IActionResult> ListItems(int categoryId, [BindRequired] int pageSize, [BindRequired] int page)
```

Comment:
`[BindRequired]` shouldn't this be `[FromQuery]`.

Reply:
With the FromQuery parameter it would be still optional.